framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -0,0 +1,672 @@
+     *
+     * @param basename the binary name of the class with the package part stripped
+     * @param aClass the AClass for {@code classname}
+     * @param printWriter the writer where the class definition should be printed
+     * @return the number of outer classes within which this class is nested
+     */
+    private static int printClassDefinitions(
+            String basename, AClassWrapper aClass, PrintWriter printWriter) {
+
+        String[] classNames = StringUtils.split(basename, '$');
+
+        for (int i = 0; i < classNames.length; i++) {
+            String nameToPrint = classNames[i];
+            printWriter.print(indents(i));
+            // For any outer class, print "class".  For a leaf class, print "enum" or "class".
+            if (i == classNames.length - 1 && aClass.isEnum()) {
+                printWriter.print("enum ");
+            } else {
+                printWriter.print("class ");
+            }
+            printWriter.print(formatAnnotations(aClass.getAnnotations()));
+            printWriter.print(nameToPrint);
+            printTypeParameters(aClass, printWriter);
+            printWriter.println(" {");
+            printWriter.println();
+        }
+        return classNames.length;
+    }
+
+    /**
+     * Prints all the fields of a given class
+     *
+     * @param aClass the class whose fields should be printed
+     * @param printWriter the writer on which to print the fields
+     * @param indentLevel the indent string
+     */
+    private static void printFields(
+            AClassWrapper aClass, PrintWriter printWriter, String indentLevel) {
+
+        if (aClass.getFields().isEmpty()) {
+            return;
+        }
+
+        printWriter.println(indentLevel + "// fields:");
+        printWriter.println();
+        for (Map.Entry<String, AFieldWrapper> fieldEntry : aClass.getFields().entrySet()) {
+            String fieldName = fieldEntry.getKey();
+            AFieldWrapper aField = fieldEntry.getValue();
+            printField(aField, fieldName, printWriter, indentLevel);
+        }
+    }
+
+    /**
+     * Prints a field declaration, including a trailing semicolon and a newline.
+     *
+     * @param aField the field declaration
+     * @param fieldName the name of the field
+     * @param printWriter the writer on which to print
+     * @param indentLevel the indent string
+     */
+    private static void printField(
+            AFieldWrapper aField, String fieldName, PrintWriter printWriter, String indentLevel) {
+        printWriter.print(indentLevel);
+        printWriter.print(formatAFieldImpl(aField, fieldName, null));
+        printWriter.println(";");
+        printWriter.println();
+    }
+
+    /**
+     * Prints a method declaration in stub file format (i.e., without a method body).
+     *
+     * @param aMethodWrapper the method to print
+     * @param basename the simple name of the containing class. Used only to determine if the method
```

Comment:
I don't see a test against `basename`.  Is the comment about how it is used accurate?
Also, this file is inconsistent about "containing class" vs "enclosing class".  Please use consistent terminology throughout.

Reply:
I can't find any other uses of "containing class" in this file (or any other in this pull request). I changed this one to "enclosing class", which I think is more precise. Can you point me to the other uses of "containing class" or similar language that you're thinking of?